src/Microsoft.IdentityModel.Validators/AadTokenValidationParametersExtension.cs

```diff
@@ -109,6 +138,45 @@ internal static bool ValidateIssuerSigningKey(SecurityKey securityKey, SecurityT
             return true;
         }
 
+        /// <summary>
+        /// Stores the cloud instance name of the signing key in a property bag and validates it.
+        /// </summary>
+        /// <param name="securityKey">The <see cref="SecurityKey"/> that signed the <see cref="SecurityToken"/>.</param>
+        /// <param name="configuration">The <see cref="BaseConfiguration"/> provided.</param>
+        /// <param name="cloudInstanceName">The cloud instance name to validate against.</param>
+        /// <returns><c>true</c> if the cloud instance name of the signing key is valid; otherwise, <c>false</c>.</returns>
+        internal static bool StoreValidateSigningKeyCloudInstanceName(SecurityKey securityKey, BaseConfiguration configuration, string cloudInstanceName)
+        {
+            if (securityKey == null)
+                return true;
+
+            if (configuration is not OpenIdConnectConfiguration openIdConnectConfiguration)
+                return true;
+
+            JsonWebKey matchedKeyFromConfig = openIdConnectConfiguration.JsonWebKeySet?.Keys.FirstOrDefault(x => x != null && x.Kid == securityKey.KeyId);
+            if (matchedKeyFromConfig != null && matchedKeyFromConfig.AdditionalData.TryGetValue(OpenIdProviderMetadataNames.CloudInstanceName, out object value))
+            {
+                string signingKeyCloudInstance = value as string;
+                if (string.IsNullOrWhiteSpace(signingKeyCloudInstance))
+                    return true;
+
+                // Store the cloud instance name in the security key's property bag.
+                securityKey.PropertyBag[OpenIdProviderMetadataNames.CloudInstanceName] = signingKeyCloudInstance;
+
+                if (cloudInstanceName == null)
+                    return true;
+
+                if (signingKeyCloudInstance != cloudInstanceName)
```

Comment:
Let's be explicit on the string comparison here (ordinal).

Reply:
fixed